ports/psoc6/main.c

```diff
@@ -60,6 +65,37 @@ void mpy_task(void *arg);
 
 TaskHandle_t mpy_task_handle;
 
+boot_mode_t check_boot_mode(void) {
+    boot_mode_t boot_mode;
+
+    // initialize user LED
+    cyhal_gpio_init(CYBSP_USER_LED, CYHAL_GPIO_DIR_OUTPUT,
+        CYHAL_GPIO_DRIVE_STRONG, CYBSP_LED_STATE_OFF);
+
+    // initialize user button
+    cyhal_gpio_init(CYBSP_USER_BTN, CYHAL_GPIO_DIR_INPUT,
+        CYHAL_GPIO_DRIVE_PULLUP, CYBSP_BTN_OFF);
+
+    if (cyhal_gpio_read(CYBSP_USER_BTN) == CYBSP_BTN_PRESSED) {
+        // Blink LED twice to indicate safe boot mode was entered
+        for (int i = 0; i < 4; i++)
+        {
+            cyhal_gpio_toggle(CYBSP_USER_LED);
+            cyhal_system_delay_ms(500); // delay in millisecond
+        }
+        boot_mode = BOOT_MODE_SAFE;
+        mp_printf(&mp_plat_print, "- DEVICE IS IN SAFE BOOT MODE -\n");
+    } else { // normal boot mode
+        boot_mode = BOOT_MODE_NORMAL;
+        mp_printf(&mp_plat_print, "- DEVICE IS IN NORMAL BOOT MODE -\n");
```

Comment:
No message is required in normal boot mode :)

Reply:
ok will remove it... :)